Handle code cache allocation when available memory becomes low

SajinaKandy opened the pull request:
When running in low memory environment like containers with limited memory, code cache allocation needs to consider the available free memory before allocating new cache. A new check to confirm safe memory limits before carving new code cache.

- [ ] OMR change:Handle code cache allocation for low physical memory [[7036](https://github.com/eclipse/omr/pull/7036)]

SajinaKandy commented:
Verbose log output:
```#CODECACHE:  The code cache repository was allocated between addresses 0000000000000000 and FFFFFFFFFFFFFFFF to be near the VM/JIT modules to avoid trampolines
#CODECACHE:  allocated code cache segment of size 268435456
#CODECACHE:  allocateCodeCacheRepository: size=268435456 heapBase=00007F96AD64C000 heapAlloc=00007F96AD64C008 heapTop=00007F96BD64C000
#CODECACHE:  Available freePhysicalMemory=11177 MB, requested codecache size=2097144 MB
#CODECACHE:  carved size=2097144 isAvailableMemorySafeForCarving=1 range: 00007F96AD64C008-00007F96AD84C000
#CODECACHE:  CodeCache allocated 00007F96CC0CBF40 @ 00007F96AD64C008-00007F96AD84C000 HelperBase:00007F96AD84B260
```
We can remove the extra verbose lines or modify them. Currently I have them for debugging in my tests.

SajinaKandy commented:
Currently this code doesn't take care of the code which doesn't use the code cache repository (old-path), the same check needs to be in the method `TR::CodeCacheMemorySegment *
J9::CodeCacheManager::allocateCodeCacheSegment(size_t segmentSize,
                                              size_t &codeCacheSizeToAllocate,
                                              void *preferredStartAddress)` before the allocation.

SajinaKandy commented:
@mpirvu Can you have a look?

mpirvu commented on runtime/compiler/runtime/J9CodeCacheManager.cpp:
```diff
@@ -76,6 +77,24 @@ J9::CodeCacheManager::initialize(bool useConsolidatedCache, uint32_t numberOfCod
+   uint64_t freePhysicalMemoryB = getCompilationInfo(_jitConfig)->computeAndCacheFreePhysicalMemory(incomplete);
+
+   if (TR::Options::isAnyVerboseOptionSet(TR_VerbosePerformance,TR_VerboseCodeCache))
+   {
```
Please fix indentation.

mpirvu commented on runtime/compiler/runtime/J9CodeCacheManager.cpp:
```diff
@@ -76,6 +77,24 @@ J9::CodeCacheManager::initialize(bool useConsolidatedCache, uint32_t numberOfCod
+   if (TR::Options::isAnyVerboseOptionSet(TR_VerbosePerformance,TR_VerboseCodeCache))
+   {
+   if (freePhysicalMemoryB != OMRPORT_MEMINFO_NOT_AVAILABLE)
+      TR_VerboseLog::writeLineLocked(TR_Vlog_CODECACHE, "Available freePhysicalMemory=%llu MB, requested codecache size=%lu MB", freePhysicalMemoryB >> 20, requestedCodeCacheSize);
```
Please use `%zu` for `size_t` variables.

mpirvu commented on runtime/compiler/runtime/J9CodeCacheManager.cpp:
```diff
@@ -76,6 +77,24 @@ J9::CodeCacheManager::initialize(bool useConsolidatedCache, uint32_t numberOfCod
    }
 
+bool
+J9::CodeCacheManager::isAvailableMemorySafeForAllocation(size_t &requestedCodeCacheSize)
```
Why do we pass `requestedCodeCacheSize` by reference?

SajinaKandy replied:
Fixed this.

mpirvu commented on runtime/compiler/runtime/J9CodeCacheManager.cpp:
```diff
@@ -76,6 +77,24 @@ J9::CodeCacheManager::initialize(bool useConsolidatedCache, uint32_t numberOfCod
+      TR_VerboseLog::writeLineLocked(TR_Vlog_CODECACHE, "Available freePhysicalMemory=%llu MB, requested codecache size=%lu MB", freePhysicalMemoryB >> 20, requestedCodeCacheSize);
+   }
+
+   if (freePhysicalMemoryB > (TR::Options::getSafeReservePhysicalMemoryValue() + requestedCodeCacheSize))
```
Maybe we should not take an action if the information about memory availability is `incomplete`

mpirvu commented on runtime/compiler/runtime/J9CodeCacheManager.cpp:
```diff
@@ -76,6 +77,24 @@ J9::CodeCacheManager::initialize(bool useConsolidatedCache, uint32_t numberOfCod
+J9::CodeCacheManager::isAvailableMemorySafeForAllocation(size_t &requestedCodeCacheSize)
+   {
+   bool incomplete;
+   uint64_t freePhysicalMemoryB = getCompilationInfo(_jitConfig)->computeAndCacheFreePhysicalMemory(incomplete);
```
The caching mechanism is used to avoid the overhead of calling this method too often. However, code caches are not allocated that frequently, so we can use `computeAndCacheFreePhysicalMemory(incompleteInfo, 0)` to specify that we want the current view of memory availability.

SajinaKandy replied:
ok. Made the changes.

mpirvu commented on runtime/compiler/runtime/J9CodeCacheManager.cpp:
```diff
@@ -76,6 +77,24 @@ J9::CodeCacheManager::initialize(bool useConsolidatedCache, uint32_t numberOfCod
+
+   if (TR::Options::isAnyVerboseOptionSet(TR_VerbosePerformance,TR_VerboseCodeCache))
+   {
+   if (freePhysicalMemoryB != OMRPORT_MEMINFO_NOT_AVAILABLE)
```
I would print this verbose information only if the allocation cannot be made, otherwise it's probably not that useful.

SajinaKandy replied:
Removed this log line from the fix

mpirvu commented on runtime/compiler/runtime/J9CodeCacheManager.hpp:
```diff
@@ -70,6 +70,7 @@ class OMR_EXTENSIBLE CodeCacheManager : public OMR::CodeCacheManagerConnector
 
    TR::CodeCache *initialize(bool useConsolidatedCache, uint32_t numberOfCodeCachesToCreateAtStartup);
 
+   bool isAvailableMemorySafeForAllocation(size_t & requestedCodeCacheSize);
```
Suggestion for the name: `isSuficientPhysicalMemoryAvailableForAllocation`

SajinaKandy commented:
In my local testing I run the Liberty Acmeair sample in a container with -m=100m to limit the memory. In order to hit the code path I use the setting `safeReservePhysicalMemoryValue=10240 -Xcodecache512K` and for testing the path without the code cache repository (old-path), I used `disableCodeCacheConsolidation`.

Lines from the test with code cache repository(default):
```
#CODECACHE:  Available freePhysicalMemory=9224192 B, requested codecache size=524288, incomplete=0, SafeReservePhysicalMemoryValue=10485760
#FAILURE:  failed to carve size=524288. Free space = 11010048. isSuficientPhysicalMemoryAvailableForAllocation = false
#FAILURE:  code cache repository exhausted
#CODECACHE:  CodeCache maximum allocated
! (AOT warm) org/jboss/weld/config/ConfigurationKey.<init>(Ljava/lang/String;ILjava/lang/String;Ljava/lang/Object;Z)V Q_SZ=6141 Q_SZI=5867 QW=36793 j9m=0000000000A40180 time=5769us compilationAotCacheFullReloFailure memLimit=262144 KB freePhysicalMemory=8 MB mem=[region=2304 system=16384]KB compThreadID=1
#PERF:  t=106745 <WARNING: JIT CACHES FULL> Disable further compilation
#FAILURE:  Method org/jboss/weld/config/ConfigurationKey.<init>(Ljava/lang/String;ILjava/lang/String;Ljava/lang/Object;Z)V will continue as interpreted
#PERF:  t=106745 <WARNING: JIT CACHES FULL> Disable further compilation
```
Lines from the test for the path without code cache repository:

```
#CODECACHE:  Available freePhysicalMemory=10788864 B, requested codecache size=524288, incomplete=0, SafeReservePhysicalMemoryValue=10485760
#CODECACHE:  Failed to allocate code cache segment due to low memory
#FAILURE:  cannot allocate code cache segment
#CODECACHE:  CodeCache maximum allocated
! (AOT warm) org/eclipse/osgi/storage/Storage.getBundleFilePath(JJ)Ljava/lang/String; Q_SZ=4215 Q_SZI=3787 QW=24930 j9m=00000000003DD0F8 time=91079us compilationCodeCacheError memLimit=262144 KB freePhysicalMemory=10 MB mem=[region=9792 system=16384]KB compThreadID=1
#FAILURE:  Method org/eclipse/osgi/storage/Storage.getBundleFilePath(JJ)Ljava/lang/String; will continue as interpreted
#PERF:  t=142156 <WARNING: JIT CACHES FULL> Disable further compilation 
```

In both paths the code cache is assumed full and further compilations are disabled.

Please note there are additional logging in the above output which were generated for testing purposes and are not there in the code change submitted for review. Once your review is complete I will run additional testing with the proposed final changes to capture the latest logs.
@mpirvu Please review.

mpirvu commented on runtime/compiler/runtime/J9CodeCacheManager.cpp:
```diff
@@ -297,6 +314,13 @@ J9::CodeCacheManager::allocateCodeCacheSegment(size_t segmentSize,
+   if (!self()->isSufficientPhysicalMemoryAvailableForAllocation(codeCacheSizeToAllocate))
+      {
+      if (config.verboseCodeCache())
+         TR_VerboseLog::writeLineLocked(TR_Vlog_CODECACHE, "Failed to allocate code cache segment due to low memory");
```
Suggestion: "...due to low physical memory"

mpirvu replied:
Another suggestion: I think it may be useful for the user to see how much we wanted to allocate and hoe much memory was available.

mpirvu commented on runtime/compiler/runtime/J9CodeCacheManager.cpp:
```diff
@@ -297,6 +314,13 @@ J9::CodeCacheManager::allocateCodeCacheSegment(size_t segmentSize,
 
+   if (!self()->isSufficientPhysicalMemoryAvailableForAllocation(codeCacheSizeToAllocate))
+      {
+      if (config.verboseCodeCache())
```
Since not being able to store compiled methods has performance implications, I would like this message to be emitted even for `verbose={compilePerformance}`

mpirvu commented on runtime/compiler/runtime/J9CodeCacheManager.cpp:
```diff
@@ -76,6 +77,28 @@ J9::CodeCacheManager::initialize(bool useConsolidatedCache, uint32_t numberOfCod
+      {
+      if (config.verboseCodeCache() || config.verbosePerformance())
+         TR_VerboseLog::writeLineLocked(TR_Vlog_CODECACHE, "Failed to allocate code cache segment due to low physical memory, requestedCodeCacheSize=%zu, freePhysicalMemoryB=%zu",
+			 		(size_t)requestedCodeCacheSize, (size_t)freePhysicalMemoryB);
```
You don't need the cast for `requestedCodeCacheSize` because it's already a `size_t` entity.

SajinaKandy commented:
@mpirvu Should we also add the `SafeReservePhysicalMemoryValue` in the below log? 
`#CODECACHE:  Failed to allocate code cache segment due to low physical memory, requestedCodeCacheSize=524288, freePhysicalMemory=10108928`

mpirvu commented:
> Should we also add the SafeReservePhysicalMemoryValue in the below log?

Yes, because someone reading the line may be baffled: requested memory is smaller than free memory.

SajinaKandy commented:
@mpirvu Regarding our discussion about looking at swap space consideration, I went through the [discussion](https://github.com/eclipse-openj9/openj9/issues/4050), particularly [comment](https://github.com/eclipse-openj9/openj9/issues/4050#issuecomment-586068351) and the subsequent [issue](https://github.com/eclipse-openj9/openj9/issues/9616).
Accordingly we set `TR_SelfTuningScratchMemoryUsageBeforeCompile` if there is no swap memory.
I do not understand what we want to do in case of Code cache memory with the swap memory check? Please can you confirm if I am looking at the right discussions and if yes then provide more guidance to the desired change.

mpirvu commented:
If there is no swap memory, then check the availability of physical memory before allocating/carving a 2 MB code cache.
If swap is present, then do nothing since the application can use the swap if it runs out of physical memory.

mpirvu commented:
I also see that https://github.com/eclipse/omr/pull/5237 is not merged, so after finishing this item you could resurrect that PR.

SajinaKandy commented:
@mpirvu Can you look at the code changes now? I am unable to yet recreate the test scenario with swap space turned off. In this case the abort compilation happens in `computeFreePhysicalLimitAndAbortCompilationIfLow`

```
 (AOT warm) Compiling jdk/internal/loader/BuiltinClassLoader.findResourceAsStream(Ljava/lang/String;Ljava/lang/String;)Ljava/io/InputStream;  OrdinaryMethod j9m=00000000000D7010  t=3329 compThreadID=1 memLimit=16442 KB freePhysicalMemory=17 MB
+ (AOT warm) java/lang/String.replace(Ljava/lang/CharSequence;Ljava/lang/CharSequence;)Ljava/lang/String; @ 00007F7D2E0FBB28-00007F7D2E0FEF48 OrdinaryMethod - Q_SZ=4852 Q_SZI=4848 QW=24492 j9m=00000000000492F8 bcsz=318 GCR time=48616us mem=[region=8832 system=16384]KB compThreadID=0 CpuLoad=0%(0%avg) JvmCpu=60% queueTime=3131552us
#PERF:  Reencoding count=250 for java/lang/String.replace(Ljava/lang/CharSequence;Ljava/lang/CharSequence;)Ljava/lang/String; j9m=00000000000492F8 
#FAILURE:  Aborting Compilation: Low On Physical Memory 14393344 B, sizeToAllocate 16777216 safeMemReserve 1484224
#FAILURE:  <EARLY TRANSLATION FAILURE: out of scratch memory>
```

In the method `computeFreePhysicalLimitAndAbortCompilationIfLow` we do not check for swap space before making a decision to abort compilations. Also all systems are likely to have swap space, in which case this new code is going to be not beneficial in handling low memory environments for docker.

mpirvu commented on runtime/compiler/runtime/J9CodeCacheManager.cpp:
```diff
@@ -76,6 +77,36 @@ J9::CodeCacheManager::initialize(bool useConsolidatedCache, uint32_t numberOfCod
+   if (TR::Options::_isSwapMemoryAvailable)
+      {
+      if (config.verboseCodeCache())
+	 TR_VerboseLog::writeLineLocked(TR_Vlog_CODECACHE, "TR::Options::_isSwapMemoryAvailable = %s", TR::Options::_isSwapMemoryAvailable? "true":"false");
```
We don't need to write this message every time a request for code cache allocations is made.

SajinaKandy replied:
True. This particular log was added for my tests purposes. Should not have been in the final fix. Will be removed.

mpirvu commented on runtime/compiler/runtime/J9CodeCacheManager.cpp:
```diff
@@ -76,6 +77,36 @@ J9::CodeCacheManager::initialize(bool useConsolidatedCache, uint32_t numberOfCod
+      return false;
+      }
+   else
+      return true;
```
Coding style: Braces around "return true" are missing

mpirvu reviewed (commented):
The fact that swap is on or off can change when CRIU is used. Thus, we need to determine again its value post restore.

SajinaKandy commented:
@mpirvu Please can you do a code review? I am yet to complete my tests to confirm the flag is set correctly post postRestore. Also need some guidance on how to recreate the code cache carving failure due to low memory. With swap memory turned off I hit the scratch memory failure as posted above. Thanks !

mpirvu commented on runtime/compiler/control/J9Options.cpp:
```diff
@@ -2653,6 +2653,7 @@ J9::Options::fePreProcess(void * base)
    if ((omrsysinfo_get_memory_info(&memInfo) == 0) && (0 == memInfo.totalSwap))
       {
       self()->setOption(TR_EnableSelfTuningScratchMemoryUsageBeforeCompile);
+      compInfo->setIsSwapMemoryDisabled();
```
If possible, I would move  the logic for determining swap availability in the constructor of CompInfo. Otherwise, there is a window of time where the new field `_isSwapMemoryDisabled` may have the incorrect value.

SajinaKandy replied:
Done.

mpirvu reviewed (commented):
One more suggestion regarding the initialization of the new field `_isSwapMemoryDisabled`

mpirvu commented:
> With swap memory turned off I hit the scratch memory failure as posted above.

I suggest you use a low value for `-Xjit:scratchSpaceLimit=`. 16384 perhaps

mpirvu commented:
@SajinaKandy  Any updates for this PR? Thanks

SajinaKandy commented:
Hi @mpirvu After I added the additional code for swap memory check in  `TR::CompilationInfo::prepareForRestore()` I wanted to test my code by disabling the swap memory after checkpoint. My additional logs in the restore was not getting generated when using the script as in the [blog1](https://blog.openj9.org/2022/09/26/getting-started-with-openj9-criu-support/) on ubuntu . Hence I then switched to a RHEL 9 machine to follow the instructions in [blog2](https://blog.openj9.org/2023/06/19/deploying-on-ocp-4-13-with-openj9-criu-support/). I could not use my idk libraries as-is because of the GLIBC version mismatch as the new scripts use UBI8 image for the checkpoint-restore tests. I then used a UBI8 image and fixed all dependency issues to build my jdk on that image to be used for testing.

The logs are now coming correctly as below:

```
#JITSTATE:  t=  3505 JIT changed state from IDLE      to STARTUP   cSmpl=  0 iSmpl=  2 comp=  7 recomp=  0, Q_SZ=  0 CLP=OFF jvmCPU=6%
#CHECKPOINT RESTORE: Done compiling methods for checkpoint
#CHECKPOINT RESTORE: Preparing to suspend threads for checkpoint
#CHECKPOINT RESTORE: Finished suspending threads for checkpoint
#CHECKPOINT RESTORE: Ready for checkpoint
#CHECKPOINT RESTORE: Preparing for restore
#CHECKPOINT RESTORE: Disabling AOT Compilation and Load
#CHECKPOINT RESTORE: Disabling AOT Compilation and Load
#CODECACHE:  At Checkpoint Restore:: Swap Memory disabled
#CHECKPOINT RESTORE: Ready for restore
#CHECKPOINT RESTORE: Start and elapsed time: startTime=1945407294, elapsedTime=  3637
#CHECKPOINT RESTORE: Reset start and elapsed time: startTime=1945791603, elapsedTime=     0
```

SajinaKandy commented on runtime/compiler/control/CompilationThread.cpp:
```diff
@@ -2946,13 +2946,23 @@ void TR::CompilationInfo::prepareForRestore()
+   // Check if there is no swap memory post restore
+   J9MemoryInfo memInfo;
+   if ((omrsysinfo_get_memory_info(&memInfo) == 0) && (0 == memInfo.totalSwap))
+   {
```
Removed this code.

SajinaKandy commented:
I tried the options `count=0,scratchSpaceLimit=16384,initialOptLevel=cold,inhibitRecompilation` in order to hit my code for low memory and get the logs printed. 
However because during compilation we check `TR::CompilationInfo::computeFreePhysicalLimitAndAbortCompilationIfLow` and abort if free physical memory is low, then in that case I think it is very unlikely for me to hit the low memory error for code cache allocation before getting the error from this method.

mpirvu commented on runtime/compiler/control/CompilationRuntime.hpp:
```diff
@@ -1045,6 +1045,9 @@ class CompilationInfo
    void              setIsInShutdownMode() {_isInShutdownMode = true;}
 
+   bool              isSwapMemoryDisabled() {return _isSwapMemoryDisabled;}
+   void              setIsSwapMemoryDisabled() {_isSwapMemoryDisabled = true;}
```
This setter should allow to set `_isSwapMemoryDisabled` to both `true` or `false`. See below why that is needed.

mpirvu commented on runtime/compiler/control/CompilationThread.cpp:
```diff
@@ -2946,6 +2948,10 @@ void TR::CompilationInfo::prepareForRestore()
    {
    J9JavaVM   *vm       = _jitConfig->javaVM;
    J9VMThread *vmThread = vm->internalVMFunctions->currentVMThread(vm);
+   TR::CompilationInfo * compInfo = getCompilationInfo(jitConfig);
```
You are already in a member of CompilationInfo; you don't need to obtain another pointer to it

mpirvu commented on runtime/compiler/control/CompilationThread.cpp:
```diff
@@ -2970,6 +2976,15 @@ void TR::CompilationInfo::prepareForRestore()
+      {
+      if (TR::Options::getCmdLineOptions()->getVerboseOption(TR_VerboseCodeCache))
+         TR_VerboseLog::writeLineLocked(TR_Vlog_CODECACHE, "At Checkpoint Restore:: Swap Memory disabled");
+      compInfo->setIsSwapMemoryDisabled();
```
If during the checkpoint we find the JVM to not use swap, but after the restore the JVM does use swap, we must revert that decision. Thus, we must use something like `setIsSwapMemoryIsDisabled(false)`. Again, you don't need to use `compInfo` because you are already in an instance of `compInfo`.

mpirvu reviewed (commented):
Please see my latest inline comments

mpirvu commented:
> However because during compilation we check TR::CompilationInfo::computeFreePhysicalLimitAndAbortCompilationIfLow and abort if free physical memory is low, then in that case I think it is very unlikely for me to hit the low memory error for code cache allocation before getting the error from this method.

I agree. You'd have to be lucky for a compilation to have enough memory to proceed and before the codegen (when the code cache is allocated) to have consumed most of the memory so that the code cache allocation fails.
For testing you might have to disable `TR::CompilationInfo::computeFreePhysicalLimitAndAbortCompilationIfLow`. You could temporarily disable this line:
```
   // enable TR_SelfTuningScratchMemoryUsageBeforeCompile if there is no swap memory
   J9MemoryInfo memInfo;
   if ((omrsysinfo_get_memory_info(&memInfo) == 0) && (0 == memInfo.totalSwap))
      {
      self()->setOption(TR_EnableSelfTuningScratchMemoryUsageBeforeCompile);
      }
```

mpirvu reviewed (approved):
LGTM

SajinaKandy commented:
Just posting the logs from various testing:

1. After commenting the scratch memory self tuning and turning off swap memory:

```
#CODECACHE:  Warning: low physical memory detected during code cache allocation, requestedCodeCacheSize=524288, freePhysicalMemory=1265664, safeMemReserve=1480192
#FAILURE:  failed to carve size=524288. Free space = 8912896. isSufficientPhysicalMemoryAvailableForAllocation = false
#FAILURE:  code cache repository exhausted
#CODECACHE:  CodeCache maximum allocated
! (cold) org/apache/aries/jndi/startup/Activator.start(Lorg/osgi/framework/BundleContext;)V Q_SZ=4 Q_SZI=4 QW=10 j9m=00000000009B5888 time=62921us compilationCodeCacheError memLimit=98304 KB freePhysicalMemory=1 MB mem=[region=1856 system=2048]KB compThreadID=0
#FAILURE:  Method org/apache/aries/jndi/startup/Activator.start(Lorg/osgi/framework/BundleContext;)V will continue as interpreted
#PERF:  t=119554 <WARNING: JIT CACHES FULL> Disable further compilation
#FAILURE:  Method com/ibm/ws/container/service/state/internal/StateChangeServiceImpl.addApplicationStateListener(Lorg/osgi/framework/ServiceReference;)V will continue as interpreted
#INFO:  Method com/ibm/ws/container/AbstractContainer.<clinit>()V will continue as interpreted
#PERF:  t=119564 <WARNING: JIT CACHES FULL> Disable further compilation
#FAILURE:  Method com/ibm/ws/channelfw/internal/ChannelFrameworkImpl.getChannelFactoryInternal(Ljava/lang/Class;Z)Lcom/ibm/wsspi/channelfw/ChannelFactory; will continue as interpreted
#PERF:  t=119564 <WARNING: JIT CACHES FULL> Disable further compilation
```

2. For the Checkpoint-restore:
```
#CHECKPOINT RESTORE: Done compiling methods for checkpoint
#CHECKPOINT RESTORE: Preparing to suspend threads for checkpoint
#CHECKPOINT RESTORE: Finished suspending threads for checkpoint
#CHECKPOINT RESTORE: Ready for checkpoint
#CHECKPOINT RESTORE: Preparing for restore
#CHECKPOINT RESTORE: Disabling AOT Compilation and Load
#CHECKPOINT RESTORE: Disabling AOT Compilation and Load
#CODECACHE:  At Checkpoint Restore:: Swap Memory is disabled
#CHECKPOINT RESTORE: Ready for restore
#CHECKPOINT RESTORE: Start and elapsed time: startTime=2290618613, elapsedTime=  3566
#CHECKPOINT RESTORE: Reset start and elapsed time: startTime=2290863909, elapsedTime=     0
```


```
#CHECKPOINT RESTORE: Done compiling methods for checkpoint
#CHECKPOINT RESTORE: Preparing to suspend threads for checkpoint
#CHECKPOINT RESTORE: Finished suspending threads for checkpoint
#CHECKPOINT RESTORE: Ready for checkpoint
#CHECKPOINT RESTORE: Preparing for restore
#CHECKPOINT RESTORE: Disabling AOT Compilation and Load
#CHECKPOINT RESTORE: Disabling AOT Compilation and Load
#CODECACHE:  At Checkpoint Restore:: Swap Memory is enabled
#CHECKPOINT RESTORE: Ready for restore
#CHECKPOINT RESTORE: Start and elapsed time: startTime=2290618613, elapsedTime=  3566
#CHECKPOINT RESTORE: Reset start and elapsed time: startTime=2291023137, elapsedTime=     0
```

mpirvu commented:
jenkins test sanity all jdk17 depends eclipse/omr#7036

mpirvu commented:
The ppc error is due to infra:
```
error: pathspec 'origin/pr/17620/merge' did not match any file(s) known to git
```

mpirvu commented:
The depending omr PR, https://github.com/eclipse/omr/pull/7036,  has reached the openj9-omr branch, so this PR can be merged.
